migrations/000032_app_versions_table.up.sql

```diff
@@ -0,0 +1,489 @@
+BEGIN;
+
+SET search_path = public, pg_catalog;
+
+--
+-- Drop obsolete views that will not work with versions.
+--
+DROP VIEW IF EXISTS app_job_types;
+DROP VIEW IF EXISTS vice_analyses;
+
+--
+-- Drop views that depend on columns which will be dropped,
+-- or that will add new view columns.
+--
+DROP VIEW IF EXISTS tool_listing;
+DROP VIEW IF EXISTS app_listing;
+DROP VIEW IF EXISTS job_listings;
+
+--
+-- app_versions table
+--
+CREATE TABLE IF NOT EXISTS app_versions (
+    id uuid NOT NULL DEFAULT uuid_generate_v1(),
+    app_id uuid NOT NULL,
+    version varchar(255) NOT NULL,
+    version_order int NOT NULL DEFAULT 0,
+    deleted boolean NOT NULL DEFAULT false,
+    disabled boolean NOT NULL DEFAULT false,
+    integration_data_id uuid NOT NULL,
+    integration_date timestamp without time zone,
+    edited_date timestamp without time zone DEFAULT CURRENT_TIMESTAMP,
+
+    PRIMARY KEY (id),
+    FOREIGN KEY (app_id) REFERENCES apps(id) ON DELETE CASCADE,
+    UNIQUE (app_id, version_order)
```

Comment:
This seems correct, I'm wondering if we should also have a `UNIQUE (app_id, version)` so an app version can't have the same version name as another version either. We might have talked about this and decided against it though, so feel free to ignore me if that's the case, haha.

Reply:
I can't remember if we already discussed this, but I'm fine with it either way.

Reply:
I thought this was a good idea after all, so I went ahead and added this unique constraint.